### PR TITLE
add utility functions to prepare input vectors

### PR DESCRIPTION
**Pull request by martingarridorc**
Those functions add the "X" and "Xmetab" labels to the input vectors and restrict them to current PKN nodes

### CODE REVIEW

**Comment by gabora** [comment]
- camel case should be replaced by _  --> at least in function names and input arguments
- add string on DESCRIPTION
- message: start with "COSMOS: "
- try to avoid shortening in input names and in function names metabData -> metabolic_data

**Comment by gabora** [comment]
pkn -> meta_network

**Comment by gabora** [comment]
remove data from inputs in `check_network_data_coverage()` function

File: R/prepare_input_vectors.R [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#'
+#' @param metabolic_data A named numeric vector, containing the values to be used for the metabolic layer in COSMOS. The names of the vector should be PUBCHEM IDs.
+#' @param meta_network Prior knowledge network created with \code{\link{load_meta_pkn()}}.
+#'
+#' @return A new vector ready to be used as COSMOS input.
+#' 
+#' @import dplyr 
+#' @importFrom stringr str_extract
+#'
+prepare_metab_data <- function(metabolic_data, meta_network) {
```
Comment (gabora) [inline R/prepare_input_vectors.R]: I would export this function as a utility

File: R/check_network_data_coverage.R [hunk 2]
```diff
@@ -34,13 +39,23 @@ check_network_data_coverage <- function(meta_network,
                     "signaling nodes from data were found in the meta PKN"))
     }
     
+    # expand metabolic input to all compartments in PKN
+    if(expand_metabolic_data) {
```
Comment (gabora) [inline R/check_network_data_coverage.R]: I would move this out as a utility function

File: R/check_network_data_coverage.R [hunk 3]
```diff
@@ -15,12 +16,16 @@
...
+#' It will add "XMetab__" before the pubchem IDs and all the possible compartments 
+#' after it. (e.g, "1150" will become "Xmetab__1150___c____" and "Xmetab__1150___e____")
 #' 
 check_network_data_coverage <- function(meta_network,
                                         tf_regulon = NULL,
                                         signaling_data,
                                         metabolic_data,
-                                        expression_data = NULL){
+                                        expression_data = NULL,
+                                        expand_metabolic_data = FALSE){
```
Comment (gabora) [inline R/check_network_data_coverage.R]: based on below comments this would be removed

**Review by gabora — COMMENTED** [review]
i checked out this version and implemneted some changes. 
More will come

**Comment by gabora** [comment]
version passed the tests.

**Comment by Nic-Nic** [comment]
Delete branch if done

**Pull request merged**